docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -201,44 +201,117 @@ The ShootState synchronization controller will become part of the gardenlet. It
 1. Normal reconciliation flow is resumed in the __Destination Seed__. Extension resources are annotated with `gardener.cloud/operation=restore` to instruct the extension controllers to reconstruct their state.
 1. The Shoot's namespace in __Source Seed__ is deleted.
 
+## Avoiding "Split Brain" Situations ("Bad Case" Scenario)
 
-### Leader Election and Control Plane Termination
+The [migration flow](#migration-workflow) described above can only be executed if both the Garden cluster and source seed cluster are healthy, and `gardenlet` in the source seed cluster can connect to the Garden cluster. In this case, `gardenlet` can directly scale down the shoot's control plane in the source seed, after checking the `spec.seedName` field.
 
-During migration "split brain" scenario must be avoided. This means that a Shoot's control plane in the __Source Seed__ must be scaled down before it is scaled up in the __Destination Seed__.
+However, there might be situations in which `gardenlet` in the source seed cluster can't connect to the Garden cluster and determine that `spec.seedName` has changed. Similarly, the connection to the seed `kube-apiserver` could also be broken. This might be caused by issues with the seed cluster itself. In other situations, the migration flow steps in the source seed might have started but might not be able to finish successfully. In all such cases, it should still be possible to migrate a shoot's control plane to a different seed, even though executing the migration flow steps in the source seed might not be possible. The potential "split brain" situation caused by having the shoot's control plane components attempting to reconcile the shoot resources in two different seeds must still be avoided, by ensuring that the shoot's control plane in the source seed is deactivated before it is activated in the destination seed.
```

Comment:
By "deactivated", I mean "stopped attempting to reconcile any of the shoot resources", effectively "switched itself off". As can be seen from the further description, this basically means:

* Shoot control plane is "down" because the `kube-apiserver" is non-functional (can't connect to `etcd`), and all controllers that rely on it are non-functional as well.
* Extension controllers are "down" for this particular shoot (its resources are not being reconciled) because its lease had expired.

I don't mention `gardenlet` because it either can't connect to garden or if it connects, it will be the first to see that the seed had changed, and stop reconciling. New snapshots are not uploaded either, by virtue of the special "final snapshot" mode introduced in `etcd-backup-restore`, as described.